### PR TITLE
Added Bookstore API

### PR DESCRIPTION
**Pull request by muneebulrehman**
## Implemented all project requirements
- Used redux-thunk
- Add book to API
- Remove book from API
- Fetch books from API on start

### CODE REVIEW

File: src/components/Booklist.jsx [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 import Input from './Input';
+import { fetchData } from '../redux/books/books';
 
 import classes from './Booklist.module.css';
 
 const Booklist = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchData());
+  }, []);
```
Comment (mmsesay) [inline src/components/Booklist.jsx:13]: - [OPTIONAL] There are some errors that come up when I start the app. Please screenshot the errors and please try to handle them correctly.

Screenshot|
:--------------:|
a1




- [OPTIONAL] I think you might need to work on the re-fetching every time someone navigates to the books component. Please note that you should fetch the data once and leave the management side to redux. Even if a visitor navigates multiple times, it should not re-fetch from the external API again. The re-fetching is causing your state to make copies of the current state on every render.

Please see the test demo to understand the point I am trying to make here.

Demo |
:-------:|
a1

**Review by mmsesay — APPROVED** [review]
Hi @muneebulrehman, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Booklist.jsx (hunk 1 above)
Comment (muneebulrehman) [reply src/components/Booklist.jsx:13]: Thanks, I noticed that after I submitted the pull request, and I already fixed it while you were reviewing the pull request. Regarding the errors, these might occur if you have prettier extension ON. But I assure you they don't appear after turning off any extension which auto formats.

Comment (mmsesay) [reply src/components/Booklist.jsx:13]: Good job in fixing them @muneebulrehman.

**Pull request merged**